vaadin-grid-flow/src/main/java/com/vaadin/flow/component/grid/Grid.java

```diff
@@ -2999,6 +3000,14 @@ public void sort(List<GridSortOrder<T>> order) {
         setSortOrder(order, false);
     }
 
+    @Override
+    protected void onAttach(AttachEvent attachEvent) {
+        super.onAttach(attachEvent);
+        if (!sortOrder.isEmpty()) {
```

Comment:
Is this check necessary? If it's only for optimization, it's probably not worth the extra lines. Logically `sortOrder` should be synchronized to the client even if it's emptied; the fact that a new Web Component instance is created on attach is an implementation detail and it's preferable for code not to rely on that.

Reply:
removed it